### ISSUE TITLE
ReorderableListView requires child keys to be identical across builds

### ISSUE THREAD

**Issue by krisgiesing** [flutter/flutter#41334]
Internally ReorderableListView creates a GlobalObjectKey per child, where that key's value is the child's key.

GlobalObjectKey uses the identical compare function to determine equality. This means that if child keys are regenerated on each build pass, the ReorderableListView subtree will rebuild, even if the keys compare equal to the previous pass.

**Comment by krisgiesing** [comment]
@Hixie per offline discussion

**Comment by krisgiesing** [comment]
Landing a fix for this is currently blocked by #43780.

**Comment by ueman** [comment]
The fix for this issue was reverted. See [here](https://github.com/flutter/flutter/commit/0e605cc4dd83137f785769dea5e8ae7da1afb361#diff-a2bb8c659b409ad45e0b796c8dc669d6). However the blocking [issue](https://github.com/flutter/flutter/issues/43780) is now resolved and the fix from [here](https://github.com/flutter/flutter/pull/41338) can reland.

**Comment by krisgiesing** [comment]
There's no technical reason I'm aware of that the fix can't move forward. The blocking issue just took a long time to resolve and in the meantime we wrote a workaround in our app so it hasn't been urgent. @gspencergoog is this issue part of the reason you were going to take a look at ReorderableListView?

**Comment by gspencergoog** [comment]
No, that wasn't the reason, but it's a good one.  My reasoning was because I need to add drag handles to it when on desktop platforms so it doesn't have to use long press there.

The current API doesn't really lend itself to that, and suffers from this issue as well, so I was looking to design a new API that would work better on all platforms.

**Comment by awhitford** [comment]
This is causing **jank** just from _selecting a list item_ -- not even reordering.
